packages/jira-adapter/src/filters/fields/context_options.ts

```diff
@@ -108,17 +109,31 @@ const updateContextOptions = async ({
   baseUrl,
   contextChange,
 }: UpdateContextOptionsParams): Promise<void> => {
-  if (addedOptions.length !== 0) {
-    const addedOptionsChunks = _.chunk(addedOptions, OPTIONS_MAXIMUM_BATCH_SIZE)
+  if (removedOptions.length !== 0) {
+    await Promise.all(removedOptions.map(async (option: Value) => {
+      await client.delete({
+        url: `${baseUrl}/${option.id}`,
+      })
+    }))
+  }
 
+  if (addedOptions.length !== 0) {
+    const optionLengthBefore = isModificationChange(contextChange)
+      ? getOptionsFromContext(contextChange.data.before).length : 0
+    // eslint-disable-next-line no-nested-ternary
+    const sliceIndex = optionLengthBefore > OPTIONS_MAXIMUM_SIZE
+      ? 0 : optionLengthBefore + addedOptions.length > OPTIONS_MAXIMUM_SIZE
+        ? OPTIONS_MAXIMUM_SIZE - optionLengthBefore : addedOptions.length
```

Comment:
nit: you can do that with max, min and remove the lint disable line
```suggestion
    const numberOfPrivateApiOptions = Math.min(
      Math.max(optionLengthBefore + addedOptions.length - OPTIONS_MAXIMUM_SIZE, 0)
      addedOptions.length
    )
```

Reply:
It should be 2 since it his case the number of private API options is 2, if you want to number of public API options you can do:
```suggestion
    const numberOfPublicApiOptions = Math.min(
      Math.max(OPTIONS_MAXIMUM_SIZE - optionLengthBefore  - addedOptions.length, 0)
      addedOptions.length
    )
```

Reply:
Yep, you right. I got confused because the slice index was the number of the public API options. Thanks :)